src/main/java/org/jenkinsci/test/acceptance/junit/AbstractATHTest.java

```diff
@@ -0,0 +1,32 @@
+package org.jenkinsci.test.acceptance.junit;
+
+import org.jenkinsci.test.acceptance.po.CapybaraPortingLayerImpl;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+/**
+ * Abstract base class factoring out common code for test classes using the {@link JenkinsAcceptanceTestRule}.
+ *
+ * @author Andres Rodriguez
+ */
+public abstract class AbstractATHTest extends CapybaraPortingLayerImpl {
+
+    protected AbstractATHTest() {
+        super(null);
+    }
+
+    /**
+     * @return finds an unused, available port on the test machine
+     */
+    public int findAvailablePort() {
```

Comment:
This should be moved to a helper class, instead. We might not need this class at all.